addons/cppcheckdata.py

```diff
@@ -1621,12 +1621,14 @@ def reportError(location, severity, message, addon, errorId, extra=''):
                 'message': message,
                 'addon': addon,
                 'errorId': errorId,
-                'extra': extra}
+                'extra': misra_severity}
```

Comment:
why don't we just use the extra. If "extra" has some text (reported from misc.py or any other addon) then why isn't that used in the output?

Reply:
I did not use extra, because today in misra.py misra_severity is stored in the extra field.
When I look in the cppcheck XML output I find following fields:
- id: misra_2012_x_y
- severity: would be ideal for the severity information
- msg: place for the error message from the file
- verbose: the place for extra information
- cwe
- file0
- location of the error

I use my own reportError function, that does write a XML-file, that is close to what cppcheck creates today. And I did not want to change the reportError function provided in cppcheckdata.py significantly to avoid any problems.

So since misra.py writes severity information twice (cppcheck_severity and misra_severity) 
https://github.com/danmar/cppcheck/blob/a87e9e104239785058e91811ea1dbf5bdc44e919/addons/misra.py#L4132

I could remove the misra_severity from misra.py, and replace that with extra or verbose information. But then I might break something when the misra checks are called through cppcheck, because then the --cli option seems to get added automatically. And the misra_severity could be used somewhere.
![image](https://github.com/danmar/cppcheck/assets/76603374/009b16f2-9d6c-4795-903b-9dfedc7e661a)

I would like to get the extra information inside, so if you can tell me how you would like me to update the code. I will try to prepare the update.

Reply:
> I use my own reportError function, that does write a XML-file, that is close to what cppcheck creates today

I do not see that here in this PR.. do you have some special function locally that writes different xml file?

Reply:
I'm using cppcheck since 1.80, and misra.py evolved a lot. I customized some checks, to not show certain issues, added custom checks, .. and to avoid merging misra.py with mine, I added a wrapper around the misra.py, and it has a XML-write feature, but I can't recall why I did that. 
I'm using the python script with the dump-files generated by cppcheck, I'm not sure if misra.py can write XML-files when used as standalone.
It was also not my intention to change anything there. I just want to add some extra/verbose information in the reports, and contribute to the tool, so it can benefit everybody.

Reply:
ok thanks